contracts/test/v0.1/RequestResponseConsumerContract.test.ts

```diff
@@ -146,10 +146,446 @@ describe('Request-Response user contract', function () {
 
     // COORDINATOR EVENT
     const fulfillEvent = coordinatorContract.interface.parseLog(fulfillReceipt.events[1])
-    expect(fulfillEvent.name).to.be.equal('DataRequestFulfilled')
+    expect(fulfillEvent.name).to.be.equal('DataRequestFulfilledUint256')
     expect(fulfillEvent.args.requestId).to.be.equal(requestId)
     expect(Number(await consumerContract.s_response())).to.be.equal(response)
   })
+
+  it('Request & Fulfill Int256', async function () {
+    const {
+      accId,
+      maxGasLimit,
+      consumerContract,
+      coordinatorContract,
+      prepaymentContract,
+      consumer,
+      rrOracle0
+    } = await loadFixture(deployFixture)
+    const prepaymentContractConsumerSigner = await ethers.getContractAt(
+      'Prepayment',
+      prepaymentContract.address,
+      consumer
+    )
+    const value = parseKlay(1)
+    await prepaymentContractConsumerSigner.deposit(accId, { value })
+    const requestReceipt = await (
+      await consumerContract.requestData(accId, maxGasLimit, {
```

Comment:
In this branch `requestData` always encodes job id using `keccak256(abi.encodePacked("Uint256"))`, so all tests should fail. Please fix the `requestData` call.

`buildRequest` inside of `RequestResponseConsumerBase` always uses `COORDINATOR.fulfillDataRequestUint256.selector` which is again wrong and the test should fail. Please fix it.

Reply:
I added a function to validate the JobID. Also, `buildRequest ` is based on appropriate `callbackfunc`
```
 function validateJobId(bytes32 jobId) internal pure returns (bool) {
        return (jobId == keccak256("uint256") ||
            jobId == keccak256("int256") ||
            jobId == keccak256("bool") ||
            jobId == keccak256("string") ||
            jobId == keccak256("bytes32") ||
            jobId == keccak256("bytes"));
    }
```

```
   function buildRequest(bytes32 jobId) internal view returns (Orakl.Request memory req) {
        bytes4 callbackFunc = detectCallbackFunc(jobId);
        return req.initialize(jobId, address(COORDINATOR), callbackFunc);
    }
```